packages/dds/tree/src/util/utils.ts

```diff
@@ -563,3 +563,49 @@ export function defineLazyCachedProperty<
 	});
 	return obj as typeof obj & { [P in K]: V };
 }
+
+/**
+ * Copies a given property from one object to another if and only if the property is present.
+ * @param source - The object to copy the property from.
+ * If `source` is undefined or does not have the given property, then this function will do nothing.
+ * @param property - The property to copy.
+ * @param destination - The object to copy the property to.
+ * @remarks This function is useful for copying properties from one object to another when the property may not be present.
```

Comment:
It would probably be worth explicitly noting that we treat a property that is _explicitly_ undefined the same way we treat a missing optional property in this context.

Reply:
Nit: Actually, given that there is nuance between copying a "defined" property and a "present" property, I wonder if it would be worth renaming this to something like `copyPropertyIfDefined` or something. Then we could add `copyPropertyIfPresent` in the future if / when the other policy is useful.

Reply:
No, you're right. I like that name better. I also like shorter names, but I think in this case the addition is meaningful and good and worth the loss of brevity. I've also clarified the docs to be consistent about what they mean by "defined" - and to spell it out in the remarks section.